improve-ephemeral-cluster-handling.md

```diff
@@ -0,0 +1,419 @@
+# Improve ephemeral cluster handling
+
+## Introduction
+
+This spec discusses ephemeral clusters in the context of Spark applications
+built using oshinko S2I builders. Spark driver applications that are
+constructed without the use of S2I are out of scope (although they could
+conceivably use the same mechanisms as long as the extended CLI is
+used to manage the cluster).
+
+Oshinko has supported ephemeral clusters for a while but currently they only
+truly work in a simple sunny-day, completed-application scenario. In the
+sunny-day scenario the spark application completes and the driver deletes the
+ephemeral cluster which it created.
+
+If a driver pod is terminated (which can happen for various reasons discussed
+below) before an application has completed, ephemeral clusters should still be
+treated appropriately depending on the circumstances. Sometimes this will mean
+deleting the cluster, and sometimes this will mean allowing the cluster to persist.
+
+Note that in particular, long-running streaming applications may never complete
+by design and therefore the intended method for stopping them is to delete the
+deploymentconfig or job that launched the driver pod. In this case the current
```

Comment:
It says later that we aren't considering jobs at the moment, so maybe I should remove job references